pkg/scheduler/objects/application.go

```diff
@@ -728,6 +730,7 @@ func (sa *Application) allocateAsk(ask *Allocation) (*resources.Resource, error)
 
 	delta := resources.Multiply(ask.GetAllocatedResource(), -1)
 	sa.pending = resources.Add(sa.pending, delta)
+	sa.pending.Prune()
 	// update the pending of the queue with the same delta
 	sa.queue.incPendingResource(delta)
```

Comment:
This construct looks weird: multiply by -1 and then call Add. We should just call `Sub` and `decPendingResource`. 
The delta that gets returned is also dropped by all callers or expected to be nil. Except for 1 tests which does not really need it. Not sure we should keep it.